migrator/src/main/scala/akka/persistence/jdbc/migrator/package.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc
+
+import akka.persistence.jdbc.config.JournalConfig
+import akka.persistence.jdbc.journal.dao.JournalQueries
+import slick.jdbc.JdbcBackend
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+package object migrator {
+
+  sealed trait DatabaseType
+
+  case object Postgres extends DatabaseType
+
+  case object MySQL extends DatabaseType
+
+  case object SqlServer extends DatabaseType
+
+  case object H2 extends DatabaseType
+
+  case object Oracle extends DatabaseType
+
+  /**
+   * helps set the new journal ordering value after the data migration is done.
+   * This trait will be implemented by the various rdbms supported by the akka persistence jdbc since
+   * the sequence number is handled differently by the various databases
+   */
+  sealed trait JournalOrdering {
```

Comment:
@Tochemey, I still need to wrap my head around this. I did tests today without looking further on the mechanism here. 

Could you explain why we need it?

Reply:
The goal is to help after the migration to reset the event_journal ordering to start with where it was in the old journal.

Reply:
Oh! I see it now.

So, the basic issue is that we are forcing new ordering because we are importing the ordering from another table and therefore, when we finished, the new table sequence/auto-increment needs to be reconfigured to start from the last + 1. Correct? 

I remember that at some point you mentioned that you were getting the ordering interleaved, but I think the main reason was because when streaming the results you were calling `sortBy(_.sequenceNumber)`.

Wouldn't we just get the right ordering if we read on one side, respecting the ordering, and insert on the other? At the end of the game, the ordering on the new table should be exactly the same. Or am I missing something obvious? 

The `JournalOrdering` you introduced looks neat if we don't have another solution. But I would first try to just ensure that the ordering is preserved by inserting on the same order. As such we will have less code.